src/components/jobManager.ts

```diff
@@ -406,4 +409,81 @@ export class JobManager {
 			uris,
 		});
 	}
+
+	public async refreshStaleJobs(storageUri: Uri): Promise<void> {
+		const staleJobs = await this.getStaleJobs();
+		const codemodsWithFilePaths: Record<string, Uri[]> = {};
+
+		staleJobs.forEach((staleJob) => {
+			if (!staleJob.oldUri) {
+				return;
+			}
+			if (!codemodsWithFilePaths[staleJob.codemodName]) {
+				codemodsWithFilePaths[staleJob.codemodName] = [];
+			}
+
+			codemodsWithFilePaths[staleJob.codemodName]?.push(staleJob.oldUri);
+		});
+
+		const executionId = buildExecutionId();
+		const happenedAt = String(Date.now());
+
+		const messages: Message[] = Object.entries(codemodsWithFilePaths).map(
+			([codemodName, filePaths]) => ({
+				kind: MessageKind.executeCodemodSet,
+				command: {
+					kind: 'executeCodemod',
+					codemodName,
+					engine: 'node',
+					storageUri,
+					uris: filePaths,
+				},
+				executionId,
+				happenedAt,
+			}),
+		);
+
+		console.log(messages);
+
+		// for (const message of messages) {
+		// 	this.#messageBus.publish(message);
+		// }
+	}
+
+	public async getStaleJobs(): Promise<Job[]> {
```

Comment:
shouldn't it be private?

Reply:
I think sometimes we may want to get stale jobs to display them somewhere